R/set_units.R

```diff
@@ -0,0 +1,52 @@
+
+
+#' set_units
+#'
+#' A pipe friendly version of \code{units<-}
+#'
+#' @param x a numeric to be assigned units, or a units object to have units
+#'   converted
+#'
+#' @param value a \code{units} object, or something coercible to one with
+#'   \code{as_units}. Depending on \code{mode}, the unit is constructed from the
+#'   supplied bare expression or from the supplied value via standard evaluation.
+#'
+#' @param ... passed on to \code{as_units}
+#'
+#' @param mode if \code{"symbols"} (the default), then unit is constructed from
+#'   the bare expression supplied. Otherwise, if\code{mode = "standard"},
+#'   standard evaluation is used for the supplied value This argument can be set
+#'   via a global option \code{options(units.set_units_mode = "standard")}
+#'
+#' @note for objects of class difftime, \code{set_units} is a simple alias for
+#'   \code{`units<-`}, and calls \code{units(x) <- value} directly without
+#'   calling \code{as_units(value)}
+#'
+#' @export
+#' @rdname set_units
+set_units <- function(x, ...) UseMethod("set_units")
+
+#' @rdname set_units
+#' @export
+set_units.default <- function(x, value, ...,
+  mode = getOption("units.set_units_mode", c("symbols", "standard"))) {
+  
+  if (missing(value))
+    value <- unitless
+  else if (match.arg(mode) == "symbols")
+    value <- substitute(value)
+  
+  if (is.null(value))
+    return(drop_units(x))
+  
+  units(x) <- as_units(value, ...)
+  x
+}
+
+#' @rdname set_units
+#' @export
+set_units.difftime <- function(x, value, ...) {
+  units(x) <- value
+  x
```

Comment:
This does not make sense, as it returns an object of class `difftime`. I think we should return `as_units(x)` rather than x (I will make this change to branch overhaul)

Reply:
My thinking here is that people will think of `set_untis` as a simple, pipe friendly alias for `units<-`. This assumption is fine to make for almost all the uses, except for the case of converting units on difftime objects.

Reply:
I think that coercion from difftime to units should be done explicitly by users, with `as_units`... `set_units` is more for assigning or converting units

Reply:
No, then you keep living in two units systems; `set_units` should move to the world of the units package. Also, the argument value needs to be character, the names incompatible with udunits and so on. Different world.

Reply:
ok. In that case, we perhaps should no longer make it a generic. I think it should error out for difftime objects, and users should be encouraged to coerce explicitly with as_units. having set_units to do automatic conversion between difftime to units feels too magical.

Reply:
I agree to remove it.